capturecat-core/src/main/java/com/capturecat/core/service/image/ImageService.java

```diff
@@ -78,10 +81,21 @@ public void addTagsToImage(Long imageId, List<String> tagNames) {
         imageTagRepository.saveAll(imageTags);
     }
 
+    @Transactional
+    public void removeTagsToImage(Long imageId, List<Long> tagIds) {
+        Image image = imageRepository.findById(imageId)
+                .orElseThrow(() -> new CoreException(ErrorType.IMAGE_NOT_FOUND));
+        List<ImageTag> imageTags = imageTagRepository.findByImageAndTagIds(image, tagIds);
+        if (imageTags.isEmpty()) {
+            return;
+        }
+        imageTagRepository.deleteAll(imageTags);
+    }
```

Comment:
```suggestion
        imageTagRepository.deleteAll(imageTags);
        
        // 태그 삭제
        // 1. ImageTag에서 Tag 조회
        List<Tag> tags = imageTags.stream().map(ImageTag::getTag).toList();
        
        // 2. 해당 태그가 삭제되고 있는지 확인 후 사용되지 않는다면 삭제
        for (Tag tag : tags) {
            if (!imageTagRepository.existsByTag(tag)) {
                tagRepository.delete(tag);
            }
        }
    }
```

대충 이런 로직이 추가될 것 같은데, 이 로직이 추가되는게 더 복잡하고 덜 직관적이지 않나요?
저는 이 로직을 추가하면 `이미지에 달린 태그 삭제` + `태그 삭제` 두 가지 역할을 한다고 생각했습니다.
사실 이 PR이 헷갈릴 수 있는게 이미지에 달린 태그를 삭제하는거라... 헷갈릴 수도 있겠네요...

Reply:
음 전 한번에 트랙잭션 처리하는 걸 선호하고, 일단 한 프로세스로 처리하고 나중에 분리의 필요성을 느끼면 분리하는 편이라
스타일이 달라서 그럴 수 있습니다. 운영 업무를 하다보니 그게 편하더라고요 ㅎㅎ